NitroxClient/MonoBehaviours/Gui/Settings/SettingsManager.cs

```diff
@@ -0,0 +1,87 @@
+﻿using NitroxClient.Unity.Helper;
+using NitroxModel.Logger;
+using Steamworks;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+namespace NitroxClient.MonoBehaviours.Gui.Settings {
+    public class SettingsManager {
+        private static string Name;
+        private static string Key_Chat;
+        private static float Color_R;
+        private static float Color_G;
+        private static float Color_B;
+        public static Image Color_Image;
+
+        public static void SetName(string name) {
+            Name = name;
+        }
+        public static string GetName() {
+            if (Name == "" || Name == null) {
+                if (SteamAPI.Init()) {
+                    Name = SteamFriends.GetPersonaName();
+                }
+            }
+            return Name;
+        }
+
+        public static void SetKey_Chat(string key) {
+            Key_Chat = key;
+        }
+        public static string GetKey_Chat() {
```

Comment:
For example `public static string ChatKey { get; set; } = "c";`.

Reply:
That doesn't work because I need a set-function for the UnityAction. Or is there a solution for this?

Reply:
I **personally** prefer a lambda expression in such cases, as to not burden the class with this kind of utility 'functions'. For instance `new UnityAction<float>((newVal) => SettingsManager.Color_R = newVal)`.